exonum/src/api/node/public/explorer.rs

```diff
@@ -297,6 +280,22 @@ impl ExplorerApi {
         });
     }
 
+    fn tx_handler(
+        state: &ServiceApiState,
+        query: TransactionHex,
+    ) -> Result<TransactionResponse, ApiError> {
+        let buf: Vec<u8> = ::hex::decode(query.tx_body).map_err(into_failure)?;
+        let signed = SignedMessage::from_raw_buffer(buf)?;
+        let tx_hash = signed.hash();
+        let signed = RawTransaction::try_from(Message::deserialize(signed)?)
+            .map_err(|_| format_err!("Couldn't deserialize transaction message."))?;
+        let _ = state
+            .sender()
+            .broadcast_transaction(signed)
+            .map_err(ApiError::from);
```

Comment:
What's the point of mapping an error if it's swallowed by `let _ = `? Also, even if the error wasn't swallowed, `broadcast_transaction` doesn't seem to return an error on oversized transaction (which I'd expect it should), which makes it difficult to have any leeway in specifying maximum acceptable JSON size.